daemon/secrets.go

```diff
@@ -1,25 +1,15 @@
 package daemon // import "github.com/docker/docker/daemon"
 
 import (
-	"context"
-
-	"github.com/containerd/log"
 	swarmtypes "github.com/docker/docker/api/types/swarm"
 )
 
 // SetContainerSecretReferences sets the container secret references needed
 func (daemon *Daemon) SetContainerSecretReferences(name string, refs []*swarmtypes.SecretReference) error {
-	if !secretsSupported() && len(refs) > 0 {
-		log.G(context.TODO()).Warn("secrets are not supported on this platform")
-		return nil
-	}
-
 	c, err := daemon.GetContainer(name)
 	if err != nil {
 		return err
 	}
-
 	c.SecretReferences = refs
```

Comment:
Interestingly, the _Configs_ implementation is slightly different here, and it _appends_, not _replaces_; https://github.com/moby/moby/blob/b0632b23457112c87b28be1850a17bfb7f9a9eac/daemon/configs.go#L21